docker/Dockerfile

```diff
@@ -0,0 +1,350 @@
+    && pip cache purge \
+    && mkdir -p $DPNP_BUILD_DIR \
+    && cd $DPNP_BUILD_DIR \
+    && git clone --recursive -b $DPNP_GIT_BRANCH --depth 1 $DPNP_GIT_URL . \
+    && source $ONEAPI_INSTALL_DIR/setvars.sh \
+    && export DPLROOT=$ONEAPI_ROOT/dpl/latest \
+    && python setup.py clean \
+    && python setup.py build_clib \
+    && export CC=dpcpp \
+    && python setup.py build_ext \
+    # XXX: is it needed to pass manylinux wheel build arg to the setup command ?
+    && python setup.py bdist_wheel \
+    && cp dist/dpnp*.whl $TMPDIR \
+    && rm -Rf $DPNP_BUILD_DIR
+
+FROM build_environment AS numba_dpex_builder
+SHELL ["/bin/bash", "-c"]
+ARG TMPDIR
+ARG ONEAPI_INSTALL_DIR
+ARG NUMBA_DPEX_BUILD_DIR=$TMPDIR/numba_dpex
+ARG NUMBA_DPEX_GIT_BRANCH
+ARG NUMBA_DPEX_GIT_URL
+ARG INTEL_PYPI_URL
+ARG BASE_PYPI_URL
+COPY --from=dpctl_builder $TMPDIR/dpctl*.whl $TMPDIR
+COPY --from=dpnp_builder $TMPDIR/dpnp*.whl $TMPDIR
+RUN pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL \
+        $TMPDIR/dpctl*.whl $TMPDIR/dpnp*.whl \
+    && pip cache purge \
+    && mkdir -p $NUMBA_DPEX_BUILD_DIR \
+    && cd $NUMBA_DPEX_BUILD_DIR \
+    && git clone --recursive -b $NUMBA_DPEX_GIT_BRANCH --depth 1 $NUMBA_DPEX_GIT_URL . \
+    && source $ONEAPI_INSTALL_DIR/setvars.sh \
+    # XXX: is it needed to pass manylinux wheel build arg to the setup command ?
+    && python setup.py bdist_wheel \
+    && cp dist/numba_dppy*.whl $TMPDIR \
+    && rm -Rf $NUMBA_DPEX_BUILD_DIR
+
+FROM base_config AS runtime_environment
+ARG ONEAPI_INSTALL_DIR
+COPY --from=python_builder $PYENV_ROOT $PYENV_ROOT
+COPY --from=python_builder $VIRTUAL_ENV $VIRTUAL_ENV
+COPY --from=oneapi_dpcpp_binaries_installer $ONEAPI_INSTALL_DIR $ONEAPI_INSTALL_DIR
+
+
+# Install the 3 packages and the spirv-tools binaries and headers
+FROM runtime_environment AS numba_dpex_dev_installer
+ARG TMPDIR
+ARG INTEL_PYPI_URL
+ARG BASE_PYPI_URL
+ARG INTEL_NUMPY_VERSION
+ARG INTEL_SCIPY_VERSION
+ARG INTEL_NUMBA_VERSION
+ARG PACKAGING_VERSION
+COPY --from=dpctl_builder $TMPDIR/dpctl*.whl $TMPDIR
+COPY --from=dpnp_builder $TMPDIR/dpnp*.whl $TMPDIR
+COPY --from=numba_dpex_builder $TMPDIR/numba_dppy*.whl $TMPDIR
+RUN pip install -U -i $INTEL_PYPI_URL "numpy${INTEL_NUMPY_VERSION}" \
+        "scipy${INTEL_SCIPY_VERSION}" --no-deps \
+    && pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL \
+        "numpy${INTEL_NUMPY_VERSION}" "numba${INTEL_NUMBA_VERSION}" \
+        "scipy${INTEL_SCIPY_VERSION}" $TMPDIR/dpctl*.whl $TMPDIR/dpnp*.whl \
+        $TMPDIR/numba_dppy*.whl packaging${PACKAGING_VERSION} \
+    && pip cache purge
+
+FROM runtime_environment AS numba_dpex_dev
+COPY --from=numba_dpex_dev_installer $VIRTUAL_ENV $VIRTUAL_ENV
+RUN apt-get update --quiet && apt-get install -y --no-install-recommends \
+        spirv-tools spirv-headers \
+    && rm -Rf /var/lib/apt/lists/*
+CMD ["/bin/bash"]
```

Comment:
We don't need it because the libraries are shipped within python packages as dependencies of intel numpy (mkl, dpcpp_rt, etc), and the path to the lib folder in the venv is added to `LD_LIBRARY_PATH` already.

The only part of the runtime that are not shipped with python packages and that are taken from the oneapi basekit (in fact, only the compiler component of the basekit) are the executables in the folder bin-llvm, that are added to the $PATH. This is hinted [here](https://github.com/IntelPython/numba-dpex#known-issue) and [here](https://github.com/IntelPython/numba-dpex/blob/main/conda-recipe/build.sh#L7).

This is the way I understand the build, but I'd advise to be cautious and to check that the jit in `numba_dpex` works as expected when using it (e.g I hope there are no such things as a silent fallback mechanism if some libraries are missing) until there's enough coverage to certify that it's the right build :)